src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(...args) {
+    if (args.length === 0) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
+
+    sum += args.reduce((acc, num) => acc + num, 0);
```

Comment:
The use of `reduce` is unnecessary since the function should only accept a single argument at a time. Instead, you should directly add the argument to the `sum` if it's a number.